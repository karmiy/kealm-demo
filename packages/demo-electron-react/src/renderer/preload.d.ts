import { IpcRendererEvent } from 'electron';
import { CHANNEL } from '../common/constants';
import { RenderMessageChannel } from '../main/renderMessageChannel';

declare global {
  interface Window {
    electron: {
      ipcRenderer: {
        sendMessage(channel: CHANNEL, ...args: unknown[]): void;
        invokeMessage(channel: CHANNEL, ...args: unknown[]): Promise<any>;
        on(
          channel: string,
          func: (_event: IpcRendererEvent, ...args: unknown[]) => void
        ): (() => void) | undefined;
        once(
          channel: string,
          func: (_event: IpcRendererEvent, ...args: unknown[]) => void
        ): void;
      };
      messageChannel: {
        get(channel: CHANNEL): RenderMessageChannel;
      };
      darkMode: {
        toggle: () => boolean;
        system: () => void;
      };
      dragger: {
        start: (fileName: string) => void;
      };
      notification: {
        show: (options: { title: string; body: string }) => void;
      };
      progressBar: {
        start: () => void;
      };
    };
  }
}

export {};
