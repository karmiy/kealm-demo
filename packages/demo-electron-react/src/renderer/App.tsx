import {
  MemoryRouter as Router,
  Routes,
  Route,
  useNavigate,
} from 'react-router-dom';
import icon from '../../assets/icon.svg';
import './App.css';
import IPC from './pages/ipc';
import MessageChannel from './pages/messageChannel';
import DarkMode from './pages/darkMode';
import Shortcut from './pages/shortcut';
import DeepLinks from './pages/deepLinks';
import DragFile from './pages/dragFile';
import Notification from './pages/notification';
import NetworkLine from './pages/networkLine';
import ProgressBar from './pages/progressBar';

const Hello = () => {
  const navigate = useNavigate();

  return (
    <div>
      <div className="Hello">
        <img width="200px" alt="icon" src={icon} />
      </div>
      <h1>electron-react-boilerplate</h1>
      <div className="Hello">
        <a
          className="m-10"
          href="https://electron-react-boilerplate.js.org/"
          target="_blank"
          rel="noreferrer"
        >
          <button type="button">
            <span role="img" aria-label="books">
              📚
            </span>
            Read Docs
          </button>
        </a>
        <a
          className="m-10"
          href="https://github.com/sponsors/electron-react-boilerplate"
          target="_blank"
          rel="noreferrer"
        >
          <button type="button">
            <span role="img" aria-label="books">
              🙏
            </span>
            Donate
          </button>
        </a>
      </div>
      <div className="Hello">
        <button className="m-10" type="button" onClick={() => navigate('ipc')}>
          <span role="img" aria-label="books" aria-hidden="true">
            🎉
          </span>
          IPC Page
        </button>
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('message-channel')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            🍕
          </span>
          Message Page
        </button>
      </div>
      <div className="Hello">
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('dark-mode')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            🎏
          </span>
          Dark Mode
        </button>
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('shortcut')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            🎄
          </span>
          Shortcut
        </button>
      </div>
      <div className="Hello">
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('deep-links')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            📙
          </span>
          Deep Links
        </button>
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('drag-file')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            📒
          </span>
          Drag File
        </button>
      </div>
      <div className="Hello">
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('notification')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            ✉️
          </span>
          Notification
        </button>
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('network-line')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            📡
          </span>
          NetworkLine
        </button>
      </div>
      <div className="Hello">
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('progress-bar')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            💿
          </span>
          ProgressBar
        </button>
      </div>
    </div>
  );
};

export default function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Hello />} />
        <Route path="/ipc" element={<IPC />} />
        <Route path="/message-channel" element={<MessageChannel />} />
        <Route path="/dark-mode" element={<DarkMode />} />
        <Route path="/shortcut" element={<Shortcut />} />
        <Route path="/deep-links" element={<DeepLinks />} />
        <Route path="/drag-file" element={<DragFile />} />
        <Route path="/notification" element={<Notification />} />
        <Route path="/network-line" element={<NetworkLine />} />
        <Route path="/progress-bar" element={<ProgressBar />} />
      </Routes>
    </Router>
  );
}
