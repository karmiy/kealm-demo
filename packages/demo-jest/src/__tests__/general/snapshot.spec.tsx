import React from 'react';
import renderer from 'react-test-renderer';

/**
 * 为什么要 snapshot 快照
 * 如果我们测试一个配置文件，如 { baseUrl: xxx, ... }，需要 expect(config).toEqual({ baseUrl: xxx, .... })
 * 每次配置文件更改都要重新调整 test 代码
 * 快照可以把我们内容映射一份结果存储起来
 * 对比前后快照，也可以更直观的看出代码或 UI 变化
 * 下次执行 test 时，与快照内容对比，不匹配报错，需要更新快照也可以进行更新
 */

// 第一次执行生成 snapshot 快照
// 之后测试 node 结构与 snapshot 是否一次
// 可以执行 jest --u / --updateSnapshot 更新 snapshot 快照
// -u 会全部更新快照，当希望快照一个个更新，可以先执行 jest --w 监听，然后会提示快照不匹配，这时控制台会提示可以输入下一步要做的，输入 i，就可以循环对每一个不匹配的项进行操作了，如每一个都输入 u 更新

/* mock names */
describe('renders correctly', () => {
    /* 测试 react 渲染节点内容是否匹配 */
    it('test innerText of wrap is "contents"', () => {
        const node = (
            <div className='wrap'>
                <p>contents</p>
            </div>
        )
        const tree = renderer
            .create(node)
            .toJSON();
        expect(tree).toMatchSnapshot();
        // expect(tree).toMatchInlineSnapshot();
    });

    /* 测试指定属性的类型是否匹配 */
    it('test createdAt is a date type and id is a number type', () => {
        const user = {
            createdAt: new Date(),
            id: Math.floor(Math.random() * 20),
            name: 'LeBron James CV',
        };

        // 匹配看是否属性类型匹配，createdAt 为 Date 类型，id 为 Number 类型
        expect(user).toMatchSnapshot({
            createdAt: expect.any(Date),
            id: expect.any(Number),
        });
    });
});
