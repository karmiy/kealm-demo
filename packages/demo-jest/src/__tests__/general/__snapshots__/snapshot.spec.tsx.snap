// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly test createdAt is a date type and id is a number type 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James CV",
}
`;

exports[`renders correctly test innerText of wrap is "contents" 1`] = `
<div
  className="wrap"
>
  <p>
    contents
  </p>
</div>
`;
